File: dsa self paced problems/Recursion/lucky number.java
class LuckyNumber
{
  public static void main(String args[])
  {
    int n=101;
    boolean res=lucky(n);
    if(res==true)
      System.out.println(n+" is lucky number.");
    else
      System.out.println(n+" is not a lucky number");
  }
  static boolean lucky(int n)
  {
    return isLucky(n,2);
  }
  static boolean isLucky(int n, int counter)
  {
    if(counter>n)
      return true;
    if(n%counter==0) // the most crucial condition that gives false result
      return false;
    return isLucky(n-n/counter, counter+1);
  }
}

File: dsa self paced problems/Recursion/lucky number.cpp
#include<iostream>
using namespace std;
int findLucky(int number, int increment)
{
    if(number==1)
        return 0;
    else
        return(findLucky(number-1, 1+increment)+increment)%number;
}
int lucky(int number)
{
    return findLucky(number,2);
}
int main()
{
  int n=5;
  cout<<lucky(n);
  return 0;
}
